website/src/models/task.js

```diff
@@ -75,6 +76,39 @@ TaskSchema.statics.sanitizeReminder = function sanitizeReminder (reminderObj) {
   return reminderObj;
 };
 
+TaskSchema.methods.scoreChallengeTask = async function scoreChallengeTask (delta) {
+  let chalTask = this;
+
+  chalTask.value += delta;
+
+  if (chalTask.type === 'habit' || chalTask.type === 'daily') {
+    // Add only one history entry per day
+    let lastChallengHistoryIndex = chalTask.history.length - 1;
+
+    if (chalTask.history[lastChallengHistoryIndex] &&
+      moment(chalTask.history[lastChallengHistoryIndex].date).isSame(new Date(), 'day')) {
+      chalTask.history[lastChallengHistoryIndex] = {
+        date: Number(new Date()),
+        value: chalTask.value,
+      };
+      chalTask.markModified(`history.${lastChallengHistoryIndex}`);
+    } else {
+      chalTask.history.push({
+        date: Number(new Date()),
+        value: chalTask.value,
+      });
+
+      // Only preen task history once a day when the task is scored first
+      if (chalTask.history.length > 365) {
+        chalTask.history = preenHistory(chalTask.history, true); // true means the challenge will retain as much entries as a subscribed user
+        chalTask.markModified(`history.${chalTask.history.length - 1}`);
```

Comment:
here the markModified is probably not necessary since the whole array is replaced, if it's needed then it should mark modified the whole array not the last item only